gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/Subscription.java

```diff
@@ -0,0 +1,150 @@
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Future;
+
+/**
+ * PubSub subscription.
+ */
+public class Subscription extends SubscriptionInfo {
+
+  private final PubSubOptions options;
+  private transient PubSub pubsub;
+
+  public static final class Builder extends SubscriptionInfo.Builder {
+
+    private final PubSub pubsub;
+    private final BuilderImpl delegate;
+
+    private Builder(Subscription subscription) {
+      pubsub = subscription.pubsub;
+      delegate = new BuilderImpl(subscription);
+    }
+
+    @Override
+    public Builder topic(String name) {
+      delegate.topic(name);
+      return this;
+    }
+
+    @Override
+    public Builder name(String name) {
+      delegate.name(name);
+      return this;
+    }
+
+    @Override
+    public Builder pushConfig(PushConfig pushConfig) {
+      delegate.pushConfig(pushConfig);
+      return this;
+    }
+
+    @Override
+    public Builder ackDeadLineSeconds(int ackDeadLineSeconds) {
+      delegate.ackDeadLineSeconds(ackDeadLineSeconds);
+      return this;
+    }
+
+    @Override
+    public Subscription build() {
+      return new Subscription(this.pubsub, this.delegate);
+    }
+  }
+
+  Subscription(PubSub pubsub, BuilderImpl builder) {
+    super(builder);
+    this.pubsub = checkNotNull(pubsub);
+    options = pubsub.options();
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(options, super.hashCode());
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+    Subscription other = (Subscription) obj;
+    return Objects.equals(toPb(), other.toPb()) && Objects.equals(options, other.options);
+  }
+
+  public PubSub pubSub() {
+    return pubsub;
+  }
+
+  public boolean delete() {
+    return pubsub.deleteSubscription(name());
+  }
+
+  public Future<Boolean> deleteAsync() {
+    return pubsub.deleteSubscriptionAsync(name());
+  }
+
+  public Subscription reload() {
+    return pubsub.getSubscription(name());
+  }
+
+  public Future<Subscription> reloadAsync() {
+    return pubsub.getSubscriptionAsync(name());
+  }
+
+  public void replacePushConfig(PushConfig pushConfig) {
+    pubsub.replacePushConfig(name(), pushConfig);
+  }
+
+  public Future<Void> replacePushConfigAsync(PushConfig pushConfig) {
+    return pubsub.replacePushConfigAsync(name(), pushConfig);
+  }
+
+  public List<ReceivedMessage> pull(PullOption... options) {
+    return pubsub.pull(name(), options);
+  }
+
+  public Future<List<Message>> pullAsync(PullOption... options) {
+    return pubsub.pullAsync(name(), options);
+  }
+
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/mziccard/gcloud-java/file/2296608819/issues/source?bid=3123109&fileBranchId=3188953#l141)